great_expectations/data_context/store/util.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from great_expectations.data_context.util import build_store_from_config
+
+logger = logging.getLogger(__name__)
+
+
+def build_in_memory_store_backend(
+    module_name: str = "great_expectations.data_context.store",
+    class_name: str = "InMemoryStoreBackend",
+    **kwargs,
+):
+    logger.debug(f"Starting data_context/store/util.py#build_in_memory_store_backend")
+    store_config: dict = {"module_name": module_name, "class_name": class_name}
+    store_config.update(**kwargs)
+    return build_store_from_config(
+        store_config=store_config, module_name=module_name, runtime_environment=None,
+    )
+
+
+def build_tuple_filesystem_store_backend(
+    base_directory: str,
+    *,
+    module_name: str = "great_expectations.data_context.store",
+    class_name: str = "TupleFilesystemStoreBackend",
+    **kwargs,
+):
+    logger.debug(
+        f"Starting data_context/store/util.py#build_tuple_filesystem_store_backend"
+    )
+    store_config: dict = {
+        "module_name": module_name,
+        "class_name": class_name,
+        "base_directory": base_directory,
+    }
+    store_config.update(**kwargs)
+    return build_store_from_config(
+        store_config=store_config, module_name=module_name, runtime_environment=None,
+    )
+
+
+def build_tuple_s3_store_backend(
+    bucket: str,
+    *,
+    module_name: str = "great_expectations.data_context.store",
+    class_name: str = "TupleS3StoreBackend",
+    **kwargs,
+):
+    logger.debug(f"Starting data_context/store/util.py#build_tuple_s3_store_backend")
+    store_config: dict = {
+        "module_name": module_name,
+        "class_name": class_name,
+        "bucket": bucket,
+    }
+    store_config.update(**kwargs)
+    return build_store_from_config(
+        store_config=store_config, module_name=module_name, runtime_environment=None,
+    )
+
+
+def build_configuration_store(
+    store_name: str,
+    store_backend,
```

Comment:
This will not work for two reasons: 1) The return type is ConfigurationStore; and 2) Type hinting here is not possible, due to circular dependencies that will ensure (we will have to refactor the entire codebase to improve this).

Reply:
Make sense to use the parent class `ConfigurationStore` :) What about trying something like this? https://stackoverflow.com/a/39757388/2916375 i.e. shielding the imports during runtime and only importing for the type checker using a pattern like:
```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from main import Main
```

Reply:
Noted; will be done in the future (ideally a refactor).